cpp/src/linear_programming/utils.cuh

```diff
@@ -129,13 +133,21 @@ struct a_divides_sqrt_b_bounded {
 };
 
 template <typename f_t>
-struct clamp {
+struct constraint_clamp {
```

Comment:
I'm not reviewing, but glanced at the changes and have a comment here: I would leave the original `clamp` untouched and introduce a new `clamp_bounds` or similar instead. This would a) be a less surprising change in the code, and b) be more consistent with C++ since the original `clamp` roughly matched the [std::clamp](https://en.cppreference.com/w/cpp/algorithm/clamp.html) syntax.